### PR TITLE
Drop python 3.8 support

### PR DESCRIPTION
**Pull request by sovrasov**
Python 3.8 is EOL is 31 Oct 2024: https://endoflife.date/python

### CODE REVIEW

File: .github/workflows/test_precommit.yml [hunk 1]
```diff
@@ -148,7 +148,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-20.04, ubuntu-22.04, macos-12]
-        python-version: [3.8, 3.9, '3.10', '3.11']
+        python-version: [3.9, '3.10', '3.11']
```
Comment (Wovchena) [inline .github/workflows/test_precommit.yml:151]: Can 3.8 be kept here and in `model_api/python/pyproject.toml`? It's not costly to keep the support

Comment (sovrasov) [reply .github/workflows/test_precommit.yml]: Yes, that's possible, but my intention was to ease type annotations in new PRs.
However, OV 2024.1 still supports 3.8, probably they'll wait till October, and then drop it. Let's keep 3.8 until that point then

Comment (Wovchena) [reply .github/workflows/test_precommit.yml]: Any feature would be a stronger reason to drop 3.8 compared to just keeping the support. You are the only users of model_api. So if you don't need 3.8, feel free to drop now.

**Review by Wovchena — APPROVED** [review]

**Comment by sovrasov** [comment]
@Wovchena I had to replace sanity tests on Win to running a sample, because of somewhat similar to https://github.com/google/googletest/issues/2628
The problem occurs even though I updated gtest commit to the latest version.

**Comment by sovrasov** [comment]
Apparently the gtest problem is still valid: https://github.com/google/googletest/issues/4556

**Review by Wovchena — APPROVED** [review]

**Pull request merged**